### PR TITLE
 Update proofOfPosession to proofOfPossession

### LINKED ISSUE #1605: Update proofOfPosession to proofOfPossession
### Actual behavior

`proofOfPossession` is incorrectly named as `proofOfPosession` (missing an `s`) within the codebase.

### Expected behavior

- `proofOfPossession` must be properly named and all the indexing behaviour must be as expected

### Steps to reproduce

- Search the codebase for `proofOfPosession`

### Which version(s) does this affect? (Environment, OS, etc...)

v0.7.0-beta.0

### PR DESCRIPTION
**Pull request by vardan10**
### What was the problem?
This PR resolves #1605 

### How was it solved?

- [x] Updated variable name `proofOfPosession` to `proofOfPossession`

### How was it tested?

1. Started lisk-core and lisk-service
2. Ran stress tests for lisk-core (stress tests registers validators and keys)
3. Queried validators table in Mysql to check proofOfPossession

### CODE REVIEW

**Comment by sameersubudhi** [comment]
@vardan10 Please extend the response for `/validator` endpoint and swagger documentation, as discussed offline.

File: services/gateway/apis/http-version3/swagger/definitions/validators.json [hunk 1]
```diff
@@ -21,6 +25,11 @@
                         "format": "id",
                         "example": "3c95f7931d61909ff092375fc8ad2bc35e393b62d5cca902",
                         "description": "BLS key"
+                    },
+                    "proofOfPossession": {
+                        "type": "string",
+                        "example": "96c5b026b1030eb73e5dfd9bfe78b0fb35e6bc7add5793fdca3d3e6a1dacb77390e998178b89f80ab8892212838bd5b2",
+                        "description": "The Proof of Possession corresponding to the BLS key."
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/validators.json:32]:
```suggestion
                        "description": "The Proof of Possession corresponding the BLS key."
```

File: services/gateway/apis/http-version3/swagger/definitions/validators.json [hunk 2]
```diff
@@ -21,6 +25,11 @@
                         "format": "id",
                         "example": "3c95f7931d61909ff092375fc8ad2bc35e393b62d5cca902",
                         "description": "BLS key"
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/validators.json:27]: Please add the `.` at the end of the descriptions for `generatorKey` as well.
```suggestion
                        "description": "BLS key."
```

**Review by sameersubudhi — APPROVED** [review]

File: services/gateway/apis/http-version3/swagger/definitions/validators.json [hunk 3]
```diff
@@ -20,7 +24,12 @@
                         "type": "string",
                         "format": "id",
                         "example": "3c95f7931d61909ff092375fc8ad2bc35e393b62d5cca902",
-                        "description": "BLS key"
+                        "description": "BLS key."
+                    },
+                    "proofOfPossession": {
+                        "type": "string",
+                        "example": "96c5b026b1030eb73e5dfd9bfe78b0fb35e6bc7add5793fdca3d3e6a1dacb77390e998178b89f80ab8892212838bd5b2",
```
Comment (nagdahimanshu) [inline services/gateway/apis/http-version3/swagger/definitions/validators.json:31]: Examples for the keys are incorrect. Can we please update the examples keys to their correct format. `generatorKey` length should be 32, `blsKey` length should be 48 and `proofOfPossession` length should be 96. What do you say @sameersubudhi ?

**Pull request merged**